rcl/src/rcl/subscription.c

```diff
@@ -72,7 +72,7 @@ rcl_subscription_init(
   rcutils_string_map_t substitutions_map = rcutils_get_zero_initialized_string_map();
   rcutils_ret_t rcutils_ret = rcutils_string_map_init(&substitutions_map, 0, rcutils_allocator);
   if (rcutils_ret != RCUTILS_RET_OK) {
-    RCL_SET_ERROR_MSG(rcutils_get_error_string().str);
+    // RCL_SET_ERROR_MSG(rcutils_get_error_string().str);
```

Comment:
@brawner @Blast545 this is something we do inconsistently throughout the code base, and while it's correct strictly speaking, it generates plenty overwrite warnings since `rcutils`, `rmw` and `rcl` error handling macros are currently the same.

Should we separate their states and propagate explicitly everywhere? Should we comment out (like here) and leave a note for future revision? Should we change `rcutils` to not complain about overwrites when the message stays the same?

Reply:
I purposefully didn't want to turn `rcutils` error state into a human-readable backtrace, as formatting is usually not appropriate (`rmw_error_msg.str` has already been formatted to provide file path and line number in a single line) and the single page char array may be exhausted rather quickly once you adopt that convention.

Whether `rcutils` error handling should support such use, that's a discussion I think we should have.

> If the warnings are important, we should try to handle the error strings when we expect them in the code (read+reset).

That's probably the right way given `rcutils` current state.

Reply:
Should we open an issue for this discussion? we'd probably need to check `rcl` at some point to remove the remaining warnings. In any case, I don't think this will block merging this PR.

Reply:
We should.

Reply:
Unless there is evidence setting the error message is causing performance issues, I don't see any need to change it. As performance tests get written, we may seem some crop up and they can be addressed on an individual basis. I've concatenated error messages before, but it's a bit tedious. The current approach of warning when there is a message overwrite has plenty of advantages actually. It let's you set the error message, but it also let's you see the propagation of error messages through the code.

Reply:
> The current approach of warning when there is a message overwrite has plenty of advantages actually. It let's you set the error message, but it also let's you see the propagation of error messages through the code.

Right, but that's not the spirit of those overwrites, which are actually `rcutils` error handling reporting there was an error with error handling itself. Also, to have libraries print to `stderr` as part of normal operation (in case a chain of errors is handled by upper layers) seems inadequate.

Reply:
I'm actually really unsure what the best solution is, but I don't think leaving commented code is desirable. I think the simplest solution would be to add a TODO with a link to the issue you create and leave the code as is. If you feel that the extra `RCL_SET_ERROR_MSG` calls should be removed, then remove them as well as adding the TODO.

Reply:
> I don't think leaving commented code is desirable

Yeah, that was intended to trigger the discussion rather than a solution for this issue.